utility/import_exporter.cpp

```diff
@@ -0,0 +1,843 @@
+		}
+	}
+	i_info->cf = cf;
+	
+	files.push_back(i_info);
+	return OK;
+}
+
+// export all the imported resources
+Error ImportExporter::export_imports(const String &output_dir){
+	String out_dir = output_dir == "" ? output_dir : GDRESettings::get_singleton()->get_project_path();
+	Error err = OK;
+	if (opt_lossy) {
+		WARN_PRINT_ONCE("Converting lossy imports, you may lose fidelity for indicated assets when re-importing upon loading the project");
+	}
+
+	for (int i = 0; i < files.size(); i++){
+		Ref<ImportInfo> iinfo = files[i];
+		String path = iinfo->get_path();
+		String source = iinfo->get_source_file();
+		String type = iinfo->get_type();
+		String importer = iinfo->importer;
+		auto loss_type = iinfo->get_import_loss_type();
+		if (loss_type != ImportInfo::LOSSLESS){
+			if (!opt_lossy){
+				lossy_imports.push_back(iinfo);
+				print_line("Not converting lossy import " + path);
+				continue;
+			}
+		}
+		if (opt_export_textures && importer == "texture"){
+			// Right now we only convert 2d image textures
+			auto tex_type = TextureLoaderCompat::recognize(path, &err);
+			switch (tex_type){
+				case TextureLoaderCompat::FORMAT_V2_IMAGE_TEXTURE:
+				case TextureLoaderCompat::FORMAT_V3_STREAM_TEXTURE2D:
+				case TextureLoaderCompat::FORMAT_V4_STREAM_TEXTURE2D: {
+					// Export texture
+					err = export_texture(out_dir, iinfo);
+				} break;
+				case TextureLoaderCompat::FORMAT_NOT_TEXTURE:
+					if (err == ERR_FILE_UNRECOGNIZED) {
+						WARN_PRINT("Import of type " + type + " is not a texture(?): " + path);
+					} else {
+						WARN_PRINT("Failed to load texture " + type + " " + path);
+					}
+					break;
+				default:
+					WARN_PRINT_ONCE("Conversion for " + type + " not yet implemented");
+					print_line("Did not convert " + type + " resource " + path);
+					not_converted.push_back(iinfo);
+					continue;
+			}
+		} else if (opt_export_samples && importer == "sample" || importer == "wav") {
+			if (iinfo->get_import_loss_type() == ImportInfo::LOSSLESS) {
+				err = export_sample(output_dir, iinfo);
+			} else {
+				// Godot doesn't support saving ADPCM samples as waves, nor converting them to PCM16
+				WARN_PRINT_ONCE("Conversion for samples stored in IMA ADPCM format not yet implemented");
+				print_line("Did not convert Sample " + path);
+				not_converted.push_back(iinfo);
+				continue;
+			}
+		} else if (opt_export_ogg && importer == "ogg_vorbis") {
+			iinfo->preferred_dest = iinfo->source_file;
+			err = convert_oggstr_to_ogg(output_dir, iinfo->import_path, iinfo->source_file);
+		} else if (opt_export_mp3 && importer == "mp3") {
+			iinfo->preferred_dest = iinfo->source_file;
+			err = convert_mp3str_to_mp3(output_dir, iinfo->import_path, iinfo->source_file);
+		} else if (opt_bin2text && iinfo->import_path.get_extension() == "res" || 
+					importer == "scene" && iinfo->source_file.get_extension() == "tscn" || 
+											iinfo->source_file.get_extension() == "escn") {
```

Comment:
Another ambiguous mix of `&&` and `||`.
```suggestion
					(importer == "scene" && (iinfo->source_file.get_extension() == "tscn" || 
											iinfo->source_file.get_extension() == "escn"))) {
```
If I understand it correctly.